extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/devconsole/DevConsoleProcessor.java

```diff
@@ -36,57 +35,100 @@ public DevConsoleTemplateInfoBuildItem collectAdditionalEndpoints(
             Path resource = i.getChildPath(StaticResourcesRecorder.META_INF_RESOURCES);
             if (resource != null && Files.exists(resource)) {
-                collectKnownPaths(resource, knownPaths);
+                collectKnownPaths(resource, staticResourceInfo);
             }
         }
 
         ClassPathUtils.consumeAsPaths(StaticResourcesRecorder.META_INF_RESOURCES, resource -> {
-            collectKnownPaths(resource, knownPaths);
+            collectKnownPaths(resource, staticResourceInfo);
         });
 
-        return new DevConsoleTemplateInfoBuildItem("staticResourcesInfo", knownPaths);
+        return new DevConsoleTemplateInfoBuildItem("staticResourcesInfo", staticResourceInfo);
     }
 
-    private void collectKnownPaths(Path resource, Set<String> staticResources) {
+    //change collectKnownPaths to build those(Resource Paths) items and fill the map
+    private void collectKnownPaths(Path resource, StaticResourceInfo staticResourceInfo) {
+
         try {
             Files.walkFileTree(resource, new SimpleFileVisitor<Path>() {
 
                 @Override
-                public FileVisitResult visitFile(Path p, BasicFileAttributes attributes) throws IOException {
-                    String simpleName = p.getFileName().toString();
-                    String file = resource.relativize(p).toString();
-                    if (simpleName.equals("index.html") || simpleName.equals("index.htm")) {
-                        Path parent = resource.relativize(p).getParent();
-                        if (parent == null) {
-                            staticResources.add("/");
+                public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs)
+                        throws IOException {
+                    Path relativePathOfDir = resource.relativize(dir);
+                    Path dirName = dir.getFileName();
+                    Path parentOfDir = dir.getParent().getFileName();
+
+                    if (!staticResourceInfo.resourceMap.containsKey(relativePathOfDir)) {
+                        StaticResourceInfo.StaticFile newStaticFile = new StaticResourceInfo.StaticFile(dirName.toString(),
+                                attrs.isDirectory(), null);
+                        staticResourceInfo.resourceMap.put(relativePathOfDir, newStaticFile);
+
+                        // add newStaticFile in list of child of parentOfDir
+                        if (staticResourceInfo.resourceMap.containsKey(parentOfDir)) {
+                            StaticResourceInfo.StaticFile valueOfParentKey = staticResourceInfo.resourceMap.get(parentOfDir);
+                            List<StaticResourceInfo.StaticFile> retrieveChildListFromParent = valueOfParentKey.children;
+                            retrieveChildListFromParent.add(newStaticFile);
+                            staticResourceInfo.resourceMap.put(parentOfDir, new StaticResourceInfo.StaticFile(
```

Comment:
Don't create a new `StaticFile`: you want to add `newStaticFile` to its parent: don't re-create it :)